### DISCUSSION TITLE
Labels

### DISCUSSION THREAD

**Discussion by CECSpecialistI** [crystalyragui/MARC2RDA#338] [Q&A] [answered]
We decided on 2022-04-06 that when we have an IRI for a value, the mapping does not need to include a label. Labels can and should be generated by implementers using IRI's, as labels can change over time. 
But, further discussion might be needed. 
What happens if/when an IRI breaks? Is it also up to implementers to cache vocabularies when they pull labels? Is worrying about broken IRI's outside our scope? 
Would it be better/more thorough to include both the label (maybe with some provenance information like a date) and the IRI when they are readily available?
Nomen?

**Comment by GordonDunsire** [comment, marked as answer]
Implementers of scenario A should be aware of this consideration, along with other Semantic Web characteristics such as AAA, OWA, and cool URIs.

The W3C Library Linked Data Incubator Group addresses it with recommendation [4.4.1 "Preserve Linked Data element sets and value vocabularies"](https://www.w3.org/2005/Incubator/lld/XGR-lld-20111025/#Preserve_Linked_Data_element_sets_and_value_vocabularies).

This is accepted as the current policy of the RDA Copyright Holders, advised by the Technical Team Liaison Officer. Thus, elements and values that have a significant change in semantics are deprecated instead of just changing labels and definitions. Deprecated IRIs continue to be included in the RDA Vocabularies and RDA Registry, although some 3R annotations may be missing.

The question of instance or dataset IRIs that reference individuals is being raised by the IFLA Cataloguing Section in the context of 'entity management' (real soon now).

I think an inappropriate label is more harmful than a broken IRI (cf recent activity about discriminatory and offensive terminology in library catalogues).

In practical terms, a 'broken' IRI is just not de-referencable for additional and human-readable data, but it still functions as an identifier and node. It might therefore be stringified and matched against identifier strings in other datasets to see if a dumb-down to implementation scenario B is found (i.e. an offline cache).

I think this is out of scope for this project. If an implementation uses ad hoc conversions, it should check the IRI immediately. If the implementation uses a bulk conversion, date-stamped, then it should take cache snapshots of de-referenced data at regular intervals and monitor the IRIs. After all, every user search in implementation scenario A has to (temporarily) cache the relevant de-referenced data.

**Reply by CECSpecialistI** [reply]
Thanks @GordonDunsire, I agree with all of this, and appreciate the heads-up about the IFLA entity management conversation. I think this is the direction we were leaning in the meeting last week if I remember correctly. If no one objects, I will mark this response as "answer" and we can move along :)
